### DISCUSSION TITLE
Interfaces missing from Device Type

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#14642] [Help Wanted!]
I've started using the public Netbox demo server to test a custom API integration.  I notice that sometimes (always?) a Device Type shows zero interfaces even though they should exist according to the Interfaces page.  I'm not sure how this problematic state was created, or how to reproduce it on my own server.  This demo server is currently running v3.7.0.  Here's one example:
The API https://demo.netbox.dev/api/dcim/device-types/7/ shows `"interface_template_count": 0`.  And the GUI page agrees: https://demo.netbox.dev/dcim/device-types/7/
<img width="1491" alt="image" src="https://github.com/netbox-community/netbox/assets/3991046/b271cc55-055c-488a-bd99-ff569a419bf0">

But the `interface-templates` API shows 51 interfaces with this device type ID.  https://demo.netbox.dev/api/dcim/interface-templates/?devicetype_id=7.  Truncated output:
```json
{
    "count": 51,
    "next": "https://demo.netbox.dev/api/dcim/interface-templates/?devicetype_id=7&limit=50&offset=50",
    "previous": null,
    "results": [
        {
            "id": 126,
            "url": "https://demo.netbox.dev/api/dcim/interface-templates/126/",
            "display": "GigabitEthernet1/0/1",
            "device_type": {
                "id": 7,
                "url": "https://demo.netbox.dev/api/dcim/device-types/7/",
                "display": "C9200-48P",
                "manufacturer": {
                    "id": 3,
                    "url": "https://demo.netbox.dev/api/dcim/manufacturers/3/",
                    "display": "Cisco",
                    "name": "Cisco",
                    "slug": "cisco"
                },
                "model": "C9200-48P",
                "slug": "c9200-48p"
            },
            "module_type": null,
            "name": "GigabitEthernet1/0/1",
            "label": "",
            "type": {
                "value": "1000base-t",
                "label": "1000BASE-T (1GE)"
            },
            "enabled": true,
            "mgmt_only": false,
            "description": "",
            "bridge": null,
            "poe_mode": null,
            "poe_type": null,
            "rf_role": null,
            "created": "2021-04-14T00:00:00Z",
            "last_updated": "2021-04-14T17:36:01.926000Z"
        },
```

After I add a new interface to this device type, now the `device-types` API https://demo.netbox.dev/api/dcim/device-types/7/ shows `"interface_template_count": 1`.

And the GUI shows all 52 interfaces:
<img width="1494" alt="image" src="https://github.com/netbox-community/netbox/assets/3991046/e4b4b098-b791-4823-9a6d-20774e831a4e">

**Comment by candlerb** [comment]
Looks to be the same problem as https://github.com/netbox-community/netbox/discussions/14430#discussioncomment-7976595 - except for device-template + interface-template rather than device + interface.

It's useful to know that the count is being initialized to zero: that would explain why the tabs are suppressed.
